### DISCUSSION TITLE
Manipulate exposure time of intermediate layers

### DISCUSSION THREAD

**Discussion by udcoperador** [sn4k3/UVtools#616] [Ideas]
i think that many people need Manipulate exposure time of intermediate layers
(sorry my english)

**Comment by sn4k3** [comment]
Can you provide a sample showing what are the intermediate layers?
Anyway you can already set exposure time per a custom layer range

**Reply by udcoperador** [reply]
![image](https://user-images.githubusercontent.com/119758244/205462754-cb9dcbcd-2bed-41b0-8bc6-be8f8995f69d.png)
A object with one or more normal exposure time.
Thanks a lot.

**Reply by sn4k3** [reply]
1. Exposure connected-sections within same layer with different times is not safe and also not possible on most of the formats. (Looks like you are looking for this?)
  - The safest and fastest method would be dim pixels at your required regions, which is already possible with UVtools but require a full capable of true AA printer.
2. Exposure disconnected objects within same layer with different times is possible but require most recent formats and firmware.
  - Fading pixels can also address this
3. Exposure whole layers with different time is possible and safe, supported by most of the recent formats.

Example of 1. and 2. with dim/fade pixels: (However you need to find the optimal value to fade, darker pixels will cure less)
![image](https://user-images.githubusercontent.com/113281/205463294-710d7237-2468-409e-b9ce-ebd2dc37e217.png)

**Reply by udcoperador** [reply]
uvtool have "fade exposure time" and "double exposure", but i cant work with that options, are in gray. im a fool

**Reply by udcoperador** [reply]
one object at diferent high

**Reply by udcoperador** [reply]
i try with your ScriptTestPerLayerSettingsSample script, Halot one not working. Same time at every layer. 5 sec.
Thanks a lot !!! ...

**Reply by udcoperador** [reply]
Congratulatons !!! . Very nice job !!!

**Reply by sn4k3** [reply]
Halot printers are 100% limited

**Comment by wolfelipe** [comment]
Trying on Creality and Elegoo but both cant change layer exposure automatically. I altered the exposure by the "Edit Print Parameteres", the first option on the tool bar.
